src/fp6.rs

```diff
@@ -21,124 +21,136 @@ use serde::de::Visitor;
 #[cfg(feature = "serialize")]
 use serde::{self, Deserialize, Deserializer, Serialize, Serializer};
 
+use crate::fp::reduce_u96;
 use crate::fp::Fp;
-use crate::fp2::Fp2;
 use crate::utils::square_assign_multi;
 
 use crate::fp::TWO_ADICITY;
 
-// 2^56 root of unity = 3537142827676469596*u + 1182629581807810213
-//                    = 1992083664070583229*u + 2727688745413696580 in Montgomery form
+const BETA: u128 = crate::fp::GENERATOR.0 as u128;
+
+// 2^33 root of unity = 10277652121819048352*u^5 + 9084844568934916810*u^4 + 6141246800624588228*u^3
+//                          + 13627025161062455919*u^2 + 2361889345279789581*u + 3733119278093849254
```

Comment:
This appears different from the value given below.

Reply:
Forgot to update this, good catch